### PR TITLE
use byteorder for endian conversion, updated rustc_version version

### PR DESCRIPTION
**Pull request by debris**

### CODE REVIEW

**Comment by debris** [comment]
@NikVolf please review

File: src/uint.rs [hunk 1]
```diff
@@ -647,55 +648,19 @@ macro_rules! construct_uint {
...
 			#[inline]
 			fn to_little_endian(&self, bytes: &mut [u8]) {
 				debug_assert!($n_words * 8 == bytes.len());
-
-				let &$name(ref arr) = self;
-				for i in 0..bytes.len() {
- 					let pos = i / 8;
- 					bytes[i] = (arr[pos] >> ((i % 8) * 8)) as u8;
+				for i in 0..$n_words {
+					(&mut bytes[8 * i..] as &mut [u8]).write_u64::<LittleEndian>(self.0[i]).expect("assert above; qed");
```
Comment (NikVolf) [inline src/uint.rs]: can be simplified as in your rlp pr

File: src/uint.rs [hunk 2]
```diff
@@ -647,55 +648,19 @@ macro_rules! construct_uint {
...
-			#[inline]
-			fn to_big_endian(&self, bytes: &mut[u8]) {
-				debug_assert!($n_words * 8 == bytes.len());
-				let &$name(ref arr) = self;
-				for i in 0..bytes.len() {
- 					let rev = bytes.len() - 1 - i;
- 					let pos = rev / 8;
- 					bytes[i] = (arr[pos] >> ((rev % 8) * 8)) as u8;
+				for i in 0..$n_words {
+					(&mut bytes[8 * i..] as &mut [u8]).write_u64::<BigEndian>(self.0[$n_words - i - 1]).expect("assert above; qed");
```
Comment (NikVolf) [inline src/uint.rs]: here as well

File: src/uint.rs (hunk 1 above)
Comment (debris) [reply src/uint.rs]: done ;)

**Pull request merged**